### DISCUSSION TITLE
Form rendering and form submitting

### DISCUSSION THREAD

**Discussion by RuslanKrupenko** [sonata-project/SonataAdminBundle#7915] [Q&A] [answered]
Hello everyone!
I was a little bit confused with CRUDController. Espesially with createAction or editAction. Can someone explaine to me why we need to call handleRequest() method of the form before checking that data was submitted? What can we handle at this moment if we just opening the page? It seems wrong for me because it breaks form events handling in Symfony framework. Such as FormEvents:PRE_SUBMIT_EVENT which is called exectly when we handle request. And if we want to modify form at this event the modifications will appeare at the page. It is not correct. For example if we add a new field that we need only with submition of the form.
Appreciate understanding and any response.

**Comment by VincentLanglet** [comment, marked as answer]
> What can we handle at this moment if we just opening the page? It seems wrong for me because it breaks form events handling in Symfony framework

There won't be any data to handle but that's a classic way to handle form.
Then you check if the form is submitted/valid.

See https://symfonycasts.com/screencast/symfony-forms/form-submit for instance.
